### PR TITLE
[HttpClient] Move configuration to PHP

### PR DESCRIPTION
**Pull request by IonBazan**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| Deprecations? | no
| Tickets       | Part of #37186
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/http_client.php [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        ->set('psr18.http_client', Psr18Client::class)
+            ->args([
+                service('http_client'),
+                service(ResponseFactoryInterface::class)->ignoreOnInvalid(),
+                service(StreamFactoryInterface::class)->ignoreOnInvalid(),
+            ])
+
+        ->alias(ClientInterface::class, 'psr18.http_client')
+
+        ->set(\Http\Client\HttpClient::class, HttplugClient::class)
```
Comment (IonBazan) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/http_client.php:44]: Not sure if FQCN here is ok but couldn't find a good alias for it.

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/http_client.php [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+        ->set('http_client', HttpClientInterface::class)
+            ->tag('monolog.logger', ['channel' => 'http_client'])
+            ->tag('http_client.client')
+            ->factory([HttpClient::class, 'create'])
+            ->args([
+                [], // default options
+                abstract_arg('max host connections'),
+            ])
+            ->call('setLogger', [service('logger')->ignoreOnInvalid()])
+            ->alias(HttpClientInterface::class, 'http_client')
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/http_client.php:33]: alias should be indented one step less (independent of `set`)

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/http_client.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+use Psr\Http\Message\StreamFactoryInterface;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\HttpClient\HttplugClient;
+use Symfony\Component\HttpClient\Psr18Client;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+return static function (ContainerConfigurator $container) {
+    $container->services()
+        ->set('http_client', HttpClientInterface::class)
+            ->tag('monolog.logger', ['channel' => 'http_client'])
```
Comment (Tobion) [inline src/Symfony/Bundle/FrameworkBundle/Resources/config/http_client.php:25]: please move tags at the end (after args/call)

File: src/Symfony/Bundle/FrameworkBundle/Resources/config/http_client.php (hunk 1 above)
Comment (Tobion) [reply src/Symfony/Bundle/FrameworkBundle/Resources/config/http_client.php:44]: seems fine

**Review by Tobion — CHANGES_REQUESTED** [review]

**Review by Tobion — APPROVED** [review]

**Comment by Tobion** [comment]
Thank you @IonBazan.

**Pull request merged**